### ISSUE TITLE
Directories missing execute bit on in-memory filesystems

### ISSUE THREAD

**Issue by lkingland** [go-git/go-git#364]
When cloning in-memory, the filesystem returned from the Worktree appears to present directories as mode 644  (unless there is an executable file in the directory... quizzical).  The directories in this Filesystem should probably be returned as 755 regardless of the modes of their contents.

These observations can be replicated by cloning `[lkingland/go-git-fs-test](https://github.com/lkingland/go-git-fs-test)` and running `go test`.

In a nutshell, it appears that the file mode of directories in the in-memory filesystem is always returning 644, which should instead probably be 755.

Thanks for writing and maintaining a very useful library!
